bin/faucet/src/server/get_tokens.rs

```diff
@@ -181,10 +188,45 @@ impl RawMintRequest {
     }
 }
 
+struct RequestCounterGuard {
+    active_count: Arc<AtomicUsize>,
+    queued_count: Arc<AtomicUsize>,
+}
+
+impl RequestCounterGuard {
+    fn new(active_count: Arc<AtomicUsize>, queued_count: Arc<AtomicUsize>) -> Self {
+        active_count.fetch_add(1, Ordering::Relaxed);
+        RequestCounterGuard { active_count, queued_count }
+    }
+
+    fn mark_queued(&self) {
+        self.queued_count.fetch_add(1, Ordering::Relaxed);
+    }
+
+    fn mark_processing(&self) {
+        self.queued_count.fetch_sub(1, Ordering::Relaxed);
+    }
+}
+
+impl Drop for RequestCounterGuard {
+    fn drop(&mut self) {
+        self.active_count.fetch_sub(1, Ordering::Relaxed);
+    }
+}
+
 pub async fn get_tokens(
     State(server): State<Server>,
     Query(request): Query<RawMintRequest>,
 ) -> Sse<impl Stream<Item = Result<Event, Infallible>>> {
+    let request_guard =
+        RequestCounterGuard::new(server.active_requests.clone(), server.queued_requests.clone());
```

Comment:
nit: You can take `&Arc<_>` in `RequestCounterGuard::new()` and clone internally for a slightly more ergonomic API here